### PR TITLE
tags: Separate versions with multidigit components with '_'

### PR DESCRIPTION
**Pull request by jwodder**
[PEP 425](https://www.python.org/dev/peps/pep-0425/) defines the version string used in Python implementation tags as follows:

> The version is *py_version_nodot*. CPython gets away with no dot, but if one is needed the underscore _ is used instead.

I take this to mean that, starting with Python 3.10, Python implementation tags will be of the form `cp3_10` instead of `cp310`.  This PR thus updates `tags.py` to insert underscores in version strings in tags whenever one or more of the version's components are at least 10.

**Note:** I am assuming that the sysconfig var `py_version_nodot` will be updated to use an underscore when Python 3.10 is released; if it is not, then the definition of `interpreter_version()` will have to be changed to use something other than `py_version_nodot`.

### CODE REVIEW

**Comment by brettcannon** [comment]
@jwodder just an FYI this PR is going to have massive merge conflicts once #231 lands (which should hopefully be in the very near future).

I also don't know yet about the proposed change. There hasn't been a discussion about how to handle Python 3.10 so I don't know how we may want to solve it. It may be that ripping out any custom version number specification for CPython is best and to just entirely rely on `nodot`, but that would require checking how compatible that is across OSs and Python versions.

**Comment by jwodder** [comment]
PR updated for #231.

File: tests/test_tags.py [hunk 1]
```diff
@@ -806,13 +955,23 @@ def test_windows_cpython(self, mock_interpreter_name, monkeypatch):
         abis = tags._cpython_abis(sys.version_info[:2])
         platforms = tags._generic_platforms()
         result = list(tags.sys_tags())
-        interpreter = "cp{major}{minor}".format(
-            major=sys.version_info[0], minor=sys.version_info[1]
+        if sys.version_info[0] >= 10 or sys.version_info[1] >= 10:
```
Comment (brettcannon) [inline tests/test_tags.py]: For the cases where it's worth keeping the logic it should probably be broken out into its own test function. Although honestly unless the test is specifically for formatting and thus has hard-coded expectations I say just call `_version_nodot()` directly.

File: tests/test_tags.py [hunk 2]
```diff
@@ -739,6 +847,39 @@ def test_default_python_version(self, monkeypatch):
             tags.Tag("py30", "none", "any"),
         ]
 
+    def test_default_python_version_needs_underscore(self, monkeypatch):
```
[the remark this answers is not in the document]
Comment (jwodder) [reply tests/test_tags.py]: It tests the same thing as `test_default_python_version()`, but using a version with a multidigit component — i.e., it tests that `sys.version_info = (3, 11)` is handled appropriately.  This completes the matrix of (default vs. explicit version) × (single-digit vs. multidigit version).

File: tests/test_tags.py [hunk 3]
```diff
@@ -543,7 +553,11 @@ def test_wide_unicode(self, unicode_size, maxunicode, version, result, monkeypat
         config = {"Py_DEBUG": 0, "WITH_PYMALLOC": 0, "Py_UNICODE_SIZE": unicode_size}
         monkeypatch.setattr(sysconfig, "get_config_var", config.__getitem__)
         monkeypatch.setattr(sys, "maxunicode", maxunicode)
-        base_abi = "cp{}{}".format(version[0], version[1])
+        if version[0] >= 10 or version[1] >= 10:
```
[the remark this answers is not in the document]
Comment (jwodder) [reply tests/test_tags.py]: The tests now use `_version_nodot()` where applicable.

**Comment by brettcannon** [comment]
@jwodder BTW when you're ready for another review just refresh the review request and it will show up in my GitHub PR review queue.

**Comment by brettcannon** [comment]
Thanks!

**Pull request merged**